src/Sportalytics.Domain/Shared/Error.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace Sportalytics.Domain.Shared;
+
+public class Error : IEquatable<Error>
+{
+    public static readonly Error None = new(string.Empty, string.Empty);
+    public static readonly Error NullValue = new("Error.NullValue", "The specified result value is null.");
+
+    public Error(string code, string message)
+    {
+        Code = code;
+        Message = message;
+    }
+
+    public string Code { get; }
+    public string Message { get; }
+    public static implicit operator string(Error error) => error.Code;
+
+    public static bool operator !=(Error? a, Error? b) => !(a == b);
```

Comment:
Are you sure you want to override all this operators?

Reply:
removed